m4/ax_check_cl.m4

```diff
@@ -1,120 +1,65 @@
-
-AC_LANG_PUSH([C])
-AX_LANG_COMPILER_MS
-AS_IF([test X$ax_compiler_ms = Xno],
-      [CL_CFLAGS="${PTHREAD_CFLAGS}"; CL_LIBS="${PTHREAD_LIBS} -lm"])
-
-#
-# Use x_includes and x_libraries if they have been set (presumably by
-# AC_PATH_X).
-#
-AS_IF([test X$no_x != Xyes],
-      [AS_IF([test -n "$x_includes"],
-             [CL_CFLAGS="-I$x_includes $CL_CFLAGS"])]
-       AS_IF([test -n "$x_libraries"],
-             [CL_LIBS="-L$x_libraries -lX11 $CL_LIBS"]))
-
-ax_save_CPPFLAGS=$CPPFLAGS
-CPPFLAGS="$CL_CFLAGS $CPPFLAGS"
-AC_CHECK_HEADERS([CL/cl.h OpenCL/cl.h])
-CPPFLAGS=$ax_save_CPPFLAGS
-
-AC_CHECK_HEADERS([windows.h])
 
-m4_define([AX_CHECK_CL_PROGRAM],
-          [AC_LANG_PROGRAM([[
-# if defined(HAVE_WINDOWS_H) && defined(_WIN32)
-#   include <windows.h>
-# endif
-# ifdef HAVE_CL_CL_H
-#   include <CL/cl.h>
-# elif defined(HAVE_OPENCL_CL_H)
-#   include <OpenCL/cl.h>
-# else
-#   error no cl.h
-# endif]],
-                           [[clFinish(0)]])])
-
-AC_CACHE_CHECK([for OpenCL library], [ax_cv_check_cl_libcl],
-[ax_cv_check_cl_libcl=no
-case $host_cpu in
-  x86_64) ax_check_cl_libdir=lib64 ;;
-  *)      ax_check_cl_libdir=lib ;;
-esac
-ax_save_CPPFLAGS=$CPPFLAGS
-CPPFLAGS="$CL_CFLAGS $CPPFLAGS"
-ax_save_LIBS=$LIBS
-LIBS=""
-ax_check_libs="-lOpenCL -lCL"
-for ax_lib in $ax_check_libs; do
-  AS_IF([test X$ax_compiler_ms = Xyes],
-        [ax_try_lib=`echo $ax_lib | $SED -e 's/^-l//' -e 's/$/.lib/'`],
-        [ax_try_lib=$ax_lib])
-  LIBS="$ax_try_lib $CL_LIBS $ax_save_LIBS"
-AC_LINK_IFELSE([AX_CHECK_CL_PROGRAM],
-               [ax_cv_check_cl_libcl=$ax_try_lib; break],
-               [ax_check_cl_nvidia_flags="-L/usr/$ax_check_cl_libdir/nvidia" LIBS="$ax_try_lib $ax_check_cl_nvidia_flags $CL_LIBS $ax_save_LIBS"
-               AC_LINK_IFELSE([AX_CHECK_CL_PROGRAM],
-                              [ax_cv_check_cl_libcl="$ax_try_lib $ax_check_cl_nvidia_flags"; break],
-                              [ax_check_cl_dylib_flag='-dylib_file /System/Library/Frameworks/OpenCL.framework/Versions/A/Libraries/libCL.dylib:/System/Library/Frameworks/OpenCL.framework/Versions/A/Libraries/libCL.dylib' LIBS="$ax_try_lib $ax_check_cl_dylib_flag $CL_LIBS $ax_save_LIBS"
-                              AC_LINK_IFELSE([AX_CHECK_CL_PROGRAM],
-                                             [ax_cv_check_cl_libcl="$ax_try_lib $ax_check_cl_dylib_flag"; break])])])
-done
-
-AS_IF([test "X$ax_cv_check_cl_libcl" = Xno -a X$no_x = Xyes],
-      [LIBS='-framework OpenCL'
-      AC_LINK_IFELSE([AX_CHECK_CL_PROGRAM],
-                     [ax_cv_check_cl_libcl=$LIBS])])
-
-LIBS=$ax_save_LIBS
-CPPFLAGS=$ax_save_CPPFLAGS])
-
-AS_IF([test "X$ax_cv_check_cl_libcl" = Xno],
-      [no_cl=yes; CL_CFLAGS=""; CL_LIBS=""],
-      [CL_LIBS="$ax_cv_check_cl_libcl $CL_LIBS"])
-AC_LANG_POP([C])
+# OpenCL support
+AC_DEFUN([AX_CHECK_CL],
+[AC_ARG_WITH([opencl],
+    [AS_HELP_STRING([--with-opencl],
+        [build OpenCL versions of SHOC benchmark programs @<:@default=check@:>@])],
```

Comment:
Needs to be adjusted I think.

Reply:
How? I am open to suggestion. 
It it the older check better?

Reply:
@ChristianBeer I add a new indentation.

Reply:
I meant for you to change the help string itself. This option toggles the opencl apps of BOINC and not the "SHOC benchmark program".

Reply:
@ChristianBeer Thank you! you are right.
I change the help text.